node/internal/url.js

```diff
@@ -0,0 +1,111 @@
+
+function domainToUnicode(domain) {
+  if (arguments.length < 1) {
+    throw new ERR_MISSING_ARGS("domain");
+  }
+
+  // toUSVString is not needed.
+  return domain;
+  // FIXME(f3n67u): implement url.domainToUnicode
+  // return url.domainToUnicode(`${domain}`);
+}
+
+const forwardSlashRegEx = /\//g;
+
+function getPathFromURLWin32(url) {
+  const hostname = url.hostname;
+  let pathname = url.pathname;
+  for (let n = 0; n < pathname.length; n++) {
+    if (pathname[n] === "%") {
+      const third = pathname.codePointAt(n + 2) | 0x20;
+      if (
+        (pathname[n + 1] === "2" && third === 102) || // 2f 2F /
+        (pathname[n + 1] === "5" && third === 99)
+      ) { // 5c 5C \
+        throw new ERR_INVALID_FILE_URL_PATH(
+          "must not include encoded \\ or / characters",
+        );
+      }
+    }
+  }
+  pathname = pathname.replace(forwardSlashRegEx, "\\");
+  pathname = decodeURIComponent(pathname);
+  if (hostname !== "") {
+    // If hostname is set, then we have a UNC path
+    // Pass the hostname through domainToUnicode just in case
+    // it is an IDN using punycode encoding. We do not need to worry
+    // about percent encoding because the URL parser will have
+    // already taken care of that for us. Note that this only
+    // causes IDNs with an appropriate `xn--` prefix to be decoded.
+    return `\\\\${domainToUnicode(hostname)}${pathname}`;
+  }
+  // Otherwise, it's a local path that requires a drive letter
+  const letter = pathname.codePointAt(1) | 0x20;
+  const sep = pathname[2];
+  if (
+    letter < CHAR_LOWERCASE_A || letter > CHAR_LOWERCASE_Z || // a..z A..Z
+    (sep !== ":")
+  ) {
+    throw new ERR_INVALID_FILE_URL_PATH("must be absolute");
+  }
+  return pathname.slice(1);
+}
+
+function getPathFromURLPosix(url) {
+  if (url.hostname !== "") {
+    throw new ERR_INVALID_FILE_URL_HOST(platform);
+  }
+  const pathname = url.pathname;
+  for (let n = 0; n < pathname.length; n++) {
+    if (pathname[n] === "%") {
+      const third = pathname.codePointAt(n + 2) | 0x20;
+      if (pathname[n + 1] === "2" && third === 102) {
+        throw new ERR_INVALID_FILE_URL_PATH(
+          "must not include encoded / characters",
+        );
+      }
+    }
+  }
+  return decodeURIComponent(pathname);
+}
+
+function fileURLToPath(path) {
+  if (typeof path === "string") {
+    path = new URL(path);
+  } else if (!isURLInstance(path)) {
+    throw new ERR_INVALID_ARG_TYPE("path", ["string", "URL"], path);
+  }
+  if (path.protocol !== "file:") {
+    throw new ERR_INVALID_URL_SCHEME("file");
+  }
+  return isWindows ? getPathFromURLWin32(path) : getPathFromURLPosix(path);
+}
+
+function isURLInstance(fileURLOrPath) {
+  return fileURLOrPath != null && fileURLOrPath.href && fileURLOrPath.origin;
+}
+
+export function toPathIfFileURL(fileURLOrPath) {
+  if (!isURLInstance(fileURLOrPath)) {
+    return fileURLOrPath;
+  }
+  return fileURLToPath(fileURLOrPath);
+}
+
+export default {
+  toPathIfFileURL,
```

Comment:
Maybe you can use here.
https://github.com/denoland/deno_std/blob/main/node/url.ts

I've implemented here with reference to https://github.com/nodejs/node/blob/master/lib/internal/url.js

Reply:
thanks for the review. I removed the `url.js` file and use `deno_std/node/url.ts` instead.